packages/components/src/FilterBar/subcomponents/FilterBarMultiSelect/FilterBarMultiSelect.tsx

```diff
@@ -39,18 +41,38 @@ const convertConsumableFormatIntoSelection = (
 
 export const FilterBarMultiSelect = ({
   id,
-  items,
+  items: propsItems,
   children,
   onSelectionChange,
   ...props
 }: FilterBarMultiSelectProps): JSX.Element | null => {
   const { getFilterState, toggleOpenFilter, updateValue, hideFilter } =
     useFilterBarContext<ConsumableSelection>()
+  const [items, setItems] = useState<ItemType[]>(propsItems)
 
   if (!id) throw Error("Missing `id` prop in FilterBarMultiSelect")
 
   const filterState = getFilterState(id)
 
+  useEffect(() => {
+    if (!checkArraysMatch(items, propsItems)) {
+      setItems(propsItems)
+    }
+  }, [propsItems])
+
+  useEffect(() => {
+    if (Array.isArray(filterState.value)) {
+      const itemValues = items.map(({ value }) => value)
+      const filteredValues = filterState.value.filter(value =>
+        itemValues.includes(value)
+      )
+
+      if (!checkArraysMatch(filterState.value, filteredValues)) {
```

Comment:
Nice, no more engineers in the building! 😉

Translation: Nice, should be a less brittle way to update those values 🕺